### ISSUE TITLE
Two-stage fallback always asks "Did you mean 'nlu_fallback'?" instead of the highest confidence user-defined intent

### ISSUE THREAD

**Issue by Devocado** [RasaHQ/rasa#7251]
**Rasa version**: 2.0.0 — 2.0.3

**Python version**: 3.8.2

**Operating system**: Ubuntu 20.04

**Issue**: Two-stage fallback always asks "Did you mean 'nlu_fallback'?" instead of the highest confidence user-defined intent

**Comment by sara-tagger** [comment]
Thanks for raising this issue, @degiz will get back to you about it soon✨

###### Please also check out the [docs](https://rasa.com/docs/) and the [forum](https://forum.rasa.com/) in case your issue was raised there too 🤗

**Comment by degiz** [comment]
Hey @Devocado 

Could you please elaborate on the issue? What does your training data look like?

**Comment by Devocado** [comment]
Hi @degiz 
This repo which I created to reproduce another bug will do: https://github.com/Devocado/rasa-two-stage-fallback-test

## Steps to reproduce:
- Train the assistant and start the interactive shell
- Type the first message as `hey, you`
The intent confidence should fail to reach the threshold, causing a fallback. The assistant will respond with
```
? Did you mean 'nlu_fallback'?  (Use arrow keys)                                                                                                                                                                   
 » 1: Yes (/nlu_fallback)
   2: No (/out_of_scope)
   Type out your own message... 
```
You will notice that I don't have an nlu_fallback intent. 
- Now open an interactive _nlu_ shell and type in the same `hey, you`
You will notice that the nlu_fallback intent is predicted, but the highest confidence user-defined intent is `greet`. 

I expect the assistant to respond to this message with "Did you mean 'greet'?" as that is the intent the message is closest to. The way that it works currently, the assistant cannot suggest any intent other that nlu_fallback, which makes that particular message pointless.

I hope this helps. Thanks

**Comment by vanshkapil** [comment]
> Hi @degiz
> This repo which I created to reproduce another bug will do: https://github.com/Devocado/rasa-two-stage-fallback-test
> 
> ## Steps to reproduce:
> * Train the assistant and start the interactive shell
> * Type the first message as `hey, you`
>   The intent confidence should fail to reach the threshold, causing a fallback. The assistant will respond with
> 
> ```
> ? Did you mean 'nlu_fallback'?  (Use arrow keys)                                                                                                                                                                   
>  » 1: Yes (/nlu_fallback)
>    2: No (/out_of_scope)
>    Type out your own message... 
> ```
> 
> You will notice that I don't have an nlu_fallback intent.
> 
> * Now open an interactive _nlu_ shell and type in the same `hey, you`
>   You will notice that the nlu_fallback intent is predicted, but the highest confidence user-defined intent is `greet`.
> 
> I expect the assistant to respond to this message with "Did you mean 'greet'?" as that is the intent the message is closest to. The way that it works currently, the assistant cannot suggest any intent other that nlu_fallback, which makes that particular message pointless.
> 
> I hope this helps. Thanks

Facing the exact same issue

**Comment by RickyvdBerg** [comment]
> **Rasa version**: 2.0.0 — 2.0.3
> 
> **Python version**: 3.8.2
> 
> **Operating system**: Ubuntu 20.04
> 
> **Issue**: Two-stage fallback always asks "Did you mean 'nlu_fallback'?" instead of the highest confidence user-defined intent

Same issue occurred on version 2.1.2. Everything is default from the "normal" example except for the language. I'm using a SpacyNLP Dutch model.

**Comment by vanshkapil** [comment]
> **Rasa version**: 2.0.0 — 2.0.3
> 
> **Python version**: 3.8.2
> 
> **Operating system**: Ubuntu 20.04
> 
> **Issue**: Two-stage fallback always asks "Did you mean 'nlu_fallback'?" instead of the highest confidence user-defined intent

I solved it by fetching the last_intent_name = tracker.latest_message.get('intent_ranking')[1].get('name')

**Comment by xinru1414** [comment]
@vanshkapil I'm really new to Rasa and facing the same issue. From my understanding the workaround happens in the action.py. Could you provide more details on how to set the intent?

**Comment by shprintuzt** [comment]
The following code worked for me.
`intent_ranking = tracker.events[-1].parse_data.get(INTENT_RANKING_KEY, [])
`

**Comment by joejuzl** [comment]
This has been fixed by #7622.
